File: c/2529.c
int maximumCount(int *nums, int numsSize)
{
  int pos = 0;
  int neg = 0;

  for (int i = 0; i < numsSize; i++)
  {

    if (nums[i] == 0)
    {
      continue;
    }
    else if (nums[i] > 0)
    {
      pos++;
    }
    else if (nums[i] < 0)
    {
      neg++;
    }
  }

  return pos > neg ? pos : neg;
}

File: python/2529.py
class Solution:
    def maximumCount(self, nums: list[int]) -> int:
        negative_numbers, positive_numbers = 0, 0
        for i in range(len(nums)):
            if nums[i] < 0:
                negative_numbers += 1
            else:
                break

        return max(negative_numbers, len(nums) - negative_numbers - nums.count(0))
